pmlc/conversion/tile_to_pxa/tile_to_pxa.cc

```diff
@@ -851,40 +851,36 @@ struct CastOpConversion : public OpConversionPattern<tile::CastOp> {
                   ConversionPatternRewriter &rewriter) const override {
     auto loc = op.getLoc();
     TileToPXATypeConverter typeConverter;
-    auto oldResultType = op.result().getType();
-    auto resultType =
-        typeConverter.convertType(oldResultType).cast<MemRefType>();
     auto operand = operands[0];
-    if (resultType == operand.getType()) {
+    if (op.result().getType() == operand.getType()) {
       rewriter.replaceOp(op, operand);
       return success();
     }
```

Comment:
It seems like canonicalization should handle this case, I think we should probably just remove it.

Reply:
SGTM; yeah I was unable to produce anything that triggered that if statement so I agree.